File: practice/17/c++/17/17/17.cpp
#include <iostream>
using namespace std;
int main() {
	setlocale(LC_ALL, "Russian");
	const int places = 37;
	int incount = 0, n = 0, black = 0, red = 0;
	int redarr[18] = { 1, 3, 5, 7, 9, 12, 14, 16, 18, 19, 21, 23, 25, 27, 30, 32, 34, 36 };
	int whole[places];
	for (int i = 0; i < places; i++) {
		whole[i] = 0;
	}
	while (true) {
		cout << "Введите номер:\n"; cin >> incount;
		if (incount >= 0) {
			whole[incount] += 1;
			for (int i = 0; i <= 18; i++) {
				if (incount == redarr[i]) {
					red += 1;
					break;
				}
				else if (i == 17) {
					black += 1;
					break;
				}
			}
		}
		else {
			break;
		}
		for (int i = 0; i < places; i++) {
			if (whole[i] > n) {
				n = whole[i];
			}
		}
		for (int i = 0; i < places; i++) {
			if (whole[i] == n) {
				cout << i << ' ';
			}
		}
		cout << endl;
		for (int i = 0; i < places; i++) {
			if (whole[i] == 0) {
				cout << i << ' ';
			}
		}
		cout << endl << red << ' ' << black << endl;
	}
}

File: practice/17/python/17.py
places = 37
whole = [0] * places
incount, n, black, red = 0, 0, 0, 0
redarr = [1, 3, 5, 7, 9, 12, 14, 16, 18, 19, 21, 23, 25, 27, 30, 32, 34, 36]
for i in range (places):
    whole[i] = 0
while True:
	incount = int(input("Введите номер:\n"))
	if incount >= 0:
		whole[incount] += 1
		for i in range (18):
			if incount == redarr[i]:
				red += 1
				break
			elif i == 17:
			    black += 1
			    break
	else:
		break
	for i in range (places):
	   	if whole[i] > n:
	   	    n = whole[i]
	for i in range (places):
		if whole[i] == n:
		    print(i, end = ' ')
	print()
	for i in range (places):
		if whole[i] == 0:
		    print(i, end = ' ')
	print('\n', red, ' ', black, '\n', sep = '')
